### ISSUE TITLE
[qa] Write test for admin network topology graph

### ISSUE THREAD

**Issue by nemesisdesign** [openwisp/docker-openwisp#98]
Write a test that wuold have catched the issue described in this thread: https://github.com/openwisp/docker-openwisp/pull/96#discussion_r372251199

**Comment by atb00ker** [comment]
I have [added a test](https://github.com/openwisp/docker-openwisp/blob/c4c98ce612f5b2d01b62a51695c99504e5f49e8f/tests/runtests.py#L139) for it, but I also wanted to check the user view by testing the `visualize.js`, but I am seeing hurdles in that:

Here is the hacky version of the final test,

```
def test_topology_visual(self):

    def _visualize_click():
        # TODO: Ensure that we open the visualize page,
        # I have tried waiting for the page to load but
        # it doesn't work, for some reason I am not able
        # to find, it doesn't work on the first click something.
        errored = False
        while not errored:
            try:
                self.base_driver.find_element_by_class_name('visualizelink').click()
            except ElementClickInterceptedException:
                errored = True

    label = 'automated-selenium-test-03'
    self.login()
    self.create_network_topology(label)
    # Test before fetch update
    self.get_resource(label, '/admin/topology/topology/', 'field-label')
    _visualize_click()
    self.assertNotEqual(-1, self.base_driver
                                .page_source
                                .find('<b>nodes</b>: <span>0</span>'))
    # Topology fetch update
    self.base_driver.get(self.config['app_url'] + '/admin/topology/topology')
    self.select_resource(label)
    self.base_driver.find_element_by_name('action') \
                    .find_element_by_xpath('//option[@value="update_selected"]') \
                    .click()
    self.base_driver.find_element_by_name('index').click()
    # Test after fetch update
    self.get_resource(label, '/admin/topology/topology/', 'field-label')
    _visualize_click()
    self.assertNotEqual(-1, self.base_driver
                                .page_source
                                .find('<b>nodes</b>: <span>23</span>'))
```

I used this because hack because `visualize` page takes time to load and and the test fails randomly at times.
I tried a couple of tricks that didn't work:
1. `time.sleep()`
2. Using selenium [wait on element](https://stackoverflow.com/a/26567563/6410464)
3. wait for `document.ready`
4. explicitly checking if `visualize.js` is loaded before running tests.
5. using `DOMAIN/topology/topology/ID` page instead.

but none of the tricks worked reliably, I get the same result of race condition and test passing half of the time. 
Unfortunately, I am out of ideas now.

**Comment by atb00ker** [comment]
This is already done! :smile:
